test/functional/api/v2/compute_resources_controller_test.rb

```diff
@@ -286,7 +288,16 @@ def teardown
   test "should not update set_console_password to true for non-VMware or non-Libvirt compute resource" do
     cr = compute_resources(:openstack)
     put :update, { :id => cr.id, :compute_resource => { :set_console_password => true } }
+    assert_response :success
     cr.reload
     assert_nil cr.attrs[:setpw]
   end
+
+  test "should not update display_type for non-Libvirt compute resource" do
+    cr = compute_resources(:openstack)
+    put :update, { :id => cr.id, :compute_resource => { :display_type => 'SPICE' } }
+    assert_response :error
```

Comment:
This is an internal server error isn't it?  Not a bad request response.  Shouldn't it have the same behaviour as set_console_password?